activerecord/lib/active_record/relation.rb

```diff
@@ -670,7 +670,17 @@ def values
     end
 
     def inspect
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      # A relation may be initialized with values that were not loaded from the
+      # database. Therefore, if there are values, they must be preserved by not
+      # loading from the database; else, use a new relation with a limit, to
+      # avoid loading an unbounded number of records into memory unnecessarily.
+      if values.any?
```

Comment:
Right, the style guide [advises](https://github.com/bbatsov/ruby-style-guide#use-if-case-returns) to “leverage the fact that if and case are expressions which return a result.”

I choose not to take the one-liner route because the `if..else..end` construct mirrors the wording of the above comment more closely.

Regarding the seemingly redundant use of `.to_a`, it evidently breaks my regression test if omitted.